crates/sui-core/src/execution_cache.rs

```diff
@@ -61,7 +61,94 @@ impl ExecutionCacheMetrics {
     }
 }
 
-pub type ExecutionCache = PassthroughCache;
+// If you have Arc<ExecutionCache>, you cannot return a reference to it as
+// an &Arc<dyn ExecutionCacheRead> (for example), because the trait object is a fat pointer.
+// So, in order to be able to return &Arc<dyn T>, we create all the converted trait objects
+// (aka fat pointers) up front and return references to them.
+#[derive(Clone)]
+pub struct ExecutionCacheTraitPointers {
+    pub cache_reader: Arc<dyn ExecutionCacheRead>,
+    pub cache_writer: Arc<dyn ExecutionCacheWrite>,
+    pub backing_store: Arc<dyn BackingStore + Send + Sync>,
+    pub backing_package_store: Arc<dyn BackingPackageStore + Send + Sync>,
+    pub object_store: Arc<dyn ObjectStore + Send + Sync>,
+    pub reconfig_api: Arc<dyn ExecutionCacheReconfigAPI>,
+    pub accumulator_store: Arc<dyn AccumulatorStore>,
+    pub checkpoint_cache: Arc<dyn CheckpointCache>,
+    pub state_sync_store: Arc<dyn StateSyncAPI>,
+    pub cache_commit: Arc<dyn ExecutionCacheCommit>,
+    pub testing_api: Arc<dyn TestingAPI>,
+}
+
+impl ExecutionCacheTraitPointers {
+    pub fn new<T>(cache: Arc<T>) -> Self
+    where
+        T: ExecutionCacheRead
+            + ExecutionCacheWrite
+            + BackingStore
+            + BackingPackageStore
+            + ObjectStore
+            + ExecutionCacheReconfigAPI
+            + AccumulatorStore
+            + CheckpointCache
+            + StateSyncAPI
+            + ExecutionCacheCommit
+            + TestingAPI
+            + 'static,
+    {
+        Self {
+            cache_reader: cache.clone(),
+            cache_writer: cache.clone(),
+            backing_store: cache.clone(),
+            backing_package_store: cache.clone(),
+            object_store: cache.clone(),
+            reconfig_api: cache.clone(),
+            accumulator_store: cache.clone(),
+            checkpoint_cache: cache.clone(),
+            state_sync_store: cache.clone(),
+            cache_commit: cache.clone(),
+            testing_api: cache.clone(),
+        }
+    }
+}
+
+static ENABLE_WRITEBACK_CACHE_ENV_VAR: &str = "ENABLE_WRITEBACK_CACHE";
```

Comment:
Why do we need both environment based and config based setting? Can we simply have a config based setting?

Reply:
It makes it very easy to run tests with the cache enabled